lib/YaST/Module.pm

```diff
@@ -0,0 +1,43 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: The class introduces methods that are common for all YaST modules.
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+package YaST::Module;
+use strict;
+use warnings;
+use testapi;
+use YuiRestClient;
+use y2_module_guitest qw(launch_yast2_module_x11);
+use y2_module_consoletest qw(yast2_console_exec);
+
+sub open {
+    my ($args) = @_;
+    my $module = $args->{module};
+    my $ui     = $args->{ui};
+    die 'No module name specified.'    unless defined $module;
+    die 'No user interface specified.' unless defined $ui;
+    if (lc($ui) eq 'ncurses') {
+        y2_module_consoletest::yast2_console_exec(
+            yast2_module => $module,
+            yast2_opts   => '--ncurses',
+            extra_vars   => get_var('YUI_PARAMS')
+        );
+    }
+    elsif (lc($ui) eq 'qt') {
+        launch_yast2_module_x11('', $module, extra_vars => get_var('YUI_PARAMS'));
+    }
+    else {
+        die "Unknown user interface: $ui";
+    }
+    YuiRestClient::connect_to_app_running_system();
```

Comment:
what happen if used for opening modules without REST API? It looks like doesn't belong here.

Reply:
Use old methods if not using libyui. All other classes that uses LibyuiClient also cannot be used without libyui. So, I hope we do not need support for this here.

Reply:
It would be nice to have it in the description, that we wait for REST API to be available, otherwise it looks like just open the module and use those methods waiting for needle but in fact additionally check that web server behind is up. I would miss that if I would not be familiar.